src/Apps/Artwork/ArtworkApp.tsx

```diff
@@ -22,46 +22,73 @@ import { track } from "Artsy/Analytics"
 import * as Schema from "Artsy/Analytics/Schema"
 import { Footer } from "Components/v2/Footer"
 import { RecentlyViewedQueryRenderer as RecentlyViewed } from "Components/v2/RecentlyViewed"
+import { RouterContext } from "found"
 import { TrackingProp } from "react-tracking"
-import { data as sd } from "sharify"
 import { get } from "Utils/get"
+import createLogger from "Utils/logger"
 import { Media } from "Utils/Responsive"
 
+const logger = createLogger("Apps/Artwork/ArtworkApp")
+
 export interface Props {
   artwork: ArtworkApp_artwork
   tracking?: TrackingProp
+  routerPathname: string
 }
 
 declare const window: any
 @track()
 export class ArtworkApp extends React.Component<Props> {
-  // TODO: Move the below tracking, which consists of:
-  //
-  //  * a custom `track` event when the artwork is acquireable or in an auction
-  //  * a custom pageview event including extra metadata
-  //
-  // into an appropriate wrapper HOC.
+  /**
+   * On mount, trigger a page view and product view
+   *
+   * FIXME: We're manually invoking pageView tracking here, instead of within
+   * the `trackingMiddleware` file as we need to pass along additional metadata.
+   * Waiting on analytics team to decide if there's a better way to capture this
+   * data that remains consistent with the rest of the app.
+   */
   componentDidMount() {
     this.trackPageview()
     this.trackProductView()
-    this.trackABTest()
   }
 
-  // TODO: Remove after AB test ends.
-  trackABTest() {
-    const { tracking } = this.props
-    const { CLIENT_NAVIGATION_V2 } = sd
+  componentDidUpdate(prevProps) {
```

Comment:
Sweet, this seems to be good locally. I recall running into an issue where the `match: { location }` prop didn't seem to update as you navigated (it always reflected the _first_ route you hit, so starting in /collect, then clicking on an artwork, and then on another artwork, would wind up being unexpected with /collect still coming thru as the prop).

Anyway! Usual caveat with `componentDidUpdate` applies, this might unexpectedly trigger more than you'd expect.

Reply:
Ah yeah, this is actually resolved by https://github.com/artsy/reaction/pull/3126/commits/c8368e1f8a16df523391e9785759d2a173dcb50d#diff-1eac5195ddbe10a2fe914a51461798d1R240-R246, which by tapping into the router context directly will always keep things perfectly in synch, as if it were a part of the middleware. (If for some reason the speediness of the analytics call remains an issue this is the direction we'd have to go.)